refactor: PrivateSend -> CoinJoin + Move the tab

xdustinface opened the pull request:
As decided by DCG this PR renames `PrivateSend` to `CoinJoin`  in the entire codebase and moves its tab to the right next to `Transactions`.

UdjinM6 reviewed (approved):
LGTM, utACK

xdustinface commented:
Pushed some more commits to rename the entire codebase.

UdjinM6 reviewed (commented):
Pls see https://github.com/UdjinM6/dash/commit/bac67e944507a746a20a6ef4658398cb59a49672 and 2 suggestions below

xdustinface commented on src/qt/dashstrings.cpp:
```diff
@@ -77,7 +77,7 @@ QT_TRANSLATE_NOOP("dash-core", ""
 "Do not keep transactions in the mempool longer than <n> hours (default: %u)"),
 QT_TRANSLATE_NOOP("dash-core", ""
-"Enable multiple PrivateSend mixing sessions per block, experimental (0-1, "
+"Enable multiple CoinJoin mixing sessions per block, experimental (0-1, "
```
oh right 👍 reverted!

xdustinface commented on src/Makefile.test.include:
```diff
@@ -110,7 +110,7 @@ BITCOIN_TESTS =\
 if ENABLE_WALLET
 BITCOIN_TESTS += \
-  test/privatesend_tests.cpp \
+  test/coinjoin_tests.cpp \
```
See 9c1a3bd

xdustinface commented:
> Pls see [UdjinM6@bac67e9](https://github.com/UdjinM6/dash/commit/bac67e944507a746a20a6ef4658398cb59a49672) and 2 suggestions below

Picked and addressed the suggestions!

UdjinM6 commented:
Found a couple more: 4dc95e29d336ca348810f04a6293d0bd0aa3db80 and a9515e3f08103d93b8762bd1be889fd7ed64cbb0

xdustinface commented:
> Found a couple more: [4dc95e2](https://github.com/dashpay/dash/commit/4dc95e29d336ca348810f04a6293d0bd0aa3db80) and [a9515e3](https://github.com/dashpay/dash/commit/a9515e3f08103d93b8762bd1be889fd7ed64cbb0)

Good catch 👍 There is also `ps_salt` but it might confuse users if they suddenly have other fully mixed amounts so if we rename it maybe like:
reads: `cj_salt` first, then tries `ps_salt` 
writes: `cj_salt`

thoughts?

UdjinM6 commented:
> There is also `ps_salt` but it might confuse users if they suddenly have other fully mixed amounts so if we rename it

Hmmm, good point... and `nCoinJoinRounds` can cause similar confusion too for users who had `nPrivateSendRounds` higher than default 🤔

xdustinface commented:
> and `nCoinJoinRounds` can cause similar confusion too for users who had `nPrivateSendRounds` higher than default 🤔

Also a good point 😄 See the last three commits, note that c92bf8928ceef787997f6aba84dbbdf116c4c634 also drops legacy checks for `nAnonymizeDashAmount`.

UdjinM6 commented:
Pls see c3110f10ef958de6f4eb1e1b3c70bef8462e930f as an alternative to b3df6df. And c92bf89 is not _that_ critical really (doesn't affect the balance), no need for that imo.

xdustinface commented:
> Pls see [c3110f1](https://github.com/dashpay/dash/commit/c3110f10ef958de6f4eb1e1b3c70bef8462e930f) as an alternative to [b3df6df](https://github.com/dashpay/dash/commit/b3df6df20d5bdfe4f79f8f5ae05bc07adb57f6a4). 

Jup thats better! But it has this e013f9c issue, no?

> And [c92bf89](https://github.com/dashpay/dash/commit/c92bf8928ceef787997f6aba84dbbdf116c4c634) is not _that_ critical really (doesn't affect the balance), no need for that imo.

Dropped but still added 74bcb8b because this could lead to unexpected denominating because default is 1k DASH if i didn't get something wrong here? 🤔

UdjinM6 commented:
> > Pls see [c3110f1](https://github.com/dashpay/dash/commit/c3110f10ef958de6f4eb1e1b3c70bef8462e930f) as an alternative to [b3df6df](https://github.com/dashpay/dash/commit/b3df6df20d5bdfe4f79f8f5ae05bc07adb57f6a4).
> 
> Jup thats better! But it has this [e013f9c](https://github.com/dashpay/dash/commit/e013f9c880115755d1d3e914786dc87cca43e4a2) issue, no?
Right, my bad 🙈 👍 


> > And [c92bf89](https://github.com/dashpay/dash/commit/c92bf8928ceef787997f6aba84dbbdf116c4c634) is not _that_ critical really (doesn't affect the balance), no need for that imo.
> 
> Dropped but still added [74bcb8b](https://github.com/dashpay/dash/commit/74bcb8b2e05ea42a837c0a3b6ba2c00420d35b52) because this could lead to unexpected denominating because default is 1k DASH if i didn't get something wrong here? 🤔

You mean in case user had it lower? True, didn't think about. Ok, let's have this one too 👍

UdjinM6 commented:
Pls see 30ec4087bfd29b35424a91684edc951d99c73b03.

Also, the more I play with it (going back to develop, saving some settings, going back) the more it feels awkward that we have some options/settings covered and some not :/ Let's maybe migrate all of them (e9b44f67b001ecbb077f5f9775ce57069aa36e0b) since we have the P-word in our code anyway now? Probably not a big issue since it's only 2 specific places and for a good reason. Thoughts?

xdustinface commented:
> Pls see [30ec408](https://github.com/dashpay/dash/commit/30ec4087bfd29b35424a91684edc951d99c73b03).

Good catch! :)

> Also, the more I play with it (going back to develop, saving some settings, going back) the more it feels awkward that we have some options/settings covered and some not :/ Let's maybe migrate all of them ([e9b44f6](https://github.com/dashpay/dash/commit/e9b44f67b001ecbb077f5f9775ce57069aa36e0b)) since we have the P-word in our code again anyway now? Probably not a big issue since it's only 2 specific places and for a good reason. Thoughts?

Well yeah, actually it just makes sense to migrate them all. IMO having the P-Word (lol for that) in there for migration shouldn't be any issue at all.. i mean, a class also have a P-Section and its still just to "remove the P-Word from older wallet databases" 😄

UdjinM6 reviewed (approved):
LGTM now :+1:

Slightly tested ACK

UdjinM6 reviewed (approved):
utACK

thephez reviewed (approved):
utACK

PastaPastaPasta reviewed (approved):
utACK
